### PR TITLE
chore: major frontend JS cleanup

### PR DESCRIPTION
**Pull request by davwheat**
**Changes proposed in this pull request:**

This PR aims to clean up our JS to utilise more standardised syntax (specifically reducing our usage of empty text nodes as opposed to rendering nothing, cleaning up class name concatenation, and using JSX more).

- Replace `(condition ? < /> : '')` with `(!!condition && < />)`
- Replace most `.component()` calls with JSX
- Clean up some string concatenation
- Use `classList` instead of concatenation
- Replace some array-based fragments with `<></>`
- Consistently use `className` instead of `class`

**Reviewers should focus on:**

**Screenshot**

**Necessity**

- [ ] Has the problem that is being solved here been clearly explained?
- [ ] If applicable, have various options for solving this problem been considered?
- [ ] For core PRs, does this need to be in core, or could it be in an extension?
- [ ] Are we willing to maintain this for years / potentially forever?

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).
- [ ] Core developer confirmed locally this works as intended.
- [ ] Tests have been added, or are not appropriate here.

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

**Review by datlechin — APPROVED** [review]

File: extensions/nicknames/js/src/forum/components/NicknameModal.js [hunk 1]
```diff
@@ -25,14 +25,11 @@ export default class NicknameModal extends Modal {
...
-                type: 'submit',
-                loading: this.loading,
-              },
-              app.translator.trans('flarum-nicknames.forum.change_nickname.submit_button')
-            )}
+            {
+              <Button className="Button Button--primary Button--block" type="submit" loading={this.loading}>
+                {app.translator.trans('flarum-nicknames.forum.change_nickname.submit_button')}
+              </Button>
+            }
```
Comment (SychO9) [inline extensions/nicknames/js/src/forum/components/NicknameModal.js]:
```suggestion
            <Button className="Button Button--primary Button--block" type="submit" loading={this.loading}>
              {app.translator.trans('flarum-nicknames.forum.change_nickname.submit_button')}
            </Button>
```

File: extensions/tags/js/src/forum/components/TagHero.js [hunk 2]
```diff
@@ -7,7 +8,7 @@ export default class TagHero extends Component {
     const color = tag.color();
 
     return (
-      <header className={'Hero TagHero' + (color ? ' TagHero--colored' : '')} style={color ? { '--hero-bg': color } : ''}>
+      <header class={classList('Hero TagHero', { 'TagHero--colored': color })} style={color ? { '--hero-bg': color } : undefined}>
```
Comment (SychO9) [inline extensions/tags/js/src/forum/components/TagHero.js]: why move away from `className` in certain places abut not others 🤔

File: extensions/tags/js/src/forum/components/TagsPage.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ import humanTime from 'flarum/common/helpers/humanTime';
 import tagIcon from '../../common/helpers/tagIcon';
 import tagLabel from '../../common/helpers/tagLabel';
 import sortTags from '../../common/utils/sortTags';
+import classList from '@flarum/core/src/common/utils/classList';
```
Comment (SychO9) [inline extensions/tags/js/src/forum/components/TagsPage.js]: for some reason even my IDE started auto importing using `@flarum/core` 🤔 
```suggestion
import classList from 'flarum/common/utils/classList';
```

File: framework/core/js/src/forum/components/NotificationList.js [hunk 4]
```diff
@@ -118,7 +118,7 @@ export default class NotificationList extends Component {
               <ul className="NotificationGroup-content">
                 {group.notifications.map((notification) => {
                   const NotificationComponent = app.notificationComponents[notification.contentType()];
-                  return NotificationComponent ? <li>{NotificationComponent.component({ notification })}</li> : '';
+                  return !!NotificationComponent && <li>{NotificationComponent.component({ notification })}</li>;
```
Comment (SychO9) [inline framework/core/js/src/forum/components/NotificationList.js]:
```suggestion
                  return !!NotificationComponent && <li><NotificationComponent notification={notification}/></li>;
```

File: framework/core/js/src/forum/components/UserCard.js [hunk 5]
```diff
@@ -31,22 +32,21 @@ export default class UserCard extends Component {
...
+                {
+                  className: 'UserCard-controls App-primaryControl',
+                  menuClassName: 'Dropdown-menu--right',
+                  buttonClassName: this.attrs.controlsButtonClassName,
+                  label: app.translator.trans('core.forum.user_controls.button'),
+                  accessibleToggleLabel: app.translator.trans('core.forum.user_controls.toggle_dropdown_accessible_label'),
+                  icon: 'fas fa-ellipsis-v',
+                },
+                controls
+              )}
```
Comment (SychO9) [inline framework/core/js/src/forum/components/UserCard.js]:
```suggestion
            {!!controls.length && (
              <Dropdown
                className="UserCard-controls App-primaryControl"
                menuClassName="Dropdown-menu--right"
                buttonClassName={this.attrs.controlsButtonClassName}
                label={app.translator.trans('core.forum.user_controls.button')}
                accessibleToggleLabel={app.translator.trans('core.forum.user_controls.toggle_dropdown_accessible_label')}
                icon="fas fa-ellipsis-v">
                {controls}
              </Dropdown>
            )}
```

**Review by SychO9 — CHANGES_REQUESTED** [review]

File: extensions/tags/js/src/forum/components/TagHero.js (hunk 2 above)
Comment (davwheat) [reply extensions/tags/js/src/forum/components/TagHero.js]: Honestly, just a case of where I could be bothered to change it...

File: extensions/tags/js/src/forum/components/TagsPage.js (hunk 3 above)
Comment (davwheat) [reply extensions/tags/js/src/forum/components/TagsPage.js]: Hmm... Strange...

File: extensions/tags/js/src/forum/components/TagHero.js (hunk 2 above)
Comment (SychO9) [reply extensions/tags/js/src/forum/components/TagHero.js]: is there a reason we'd want to move towards just using `class`?

Comment (davwheat) [reply extensions/tags/js/src/forum/components/TagHero.js]: Smaller bundle size? 😁

**Review by SychO9 — APPROVED** [review]

**Comment by SychO9** [comment]
tests failing 🤔 hard to tell but maybe formatting?

**Comment by davwheat** [comment]
I've run `yarn workspaces run format` and `yarn workspaces run build` and both work fine... This is weird...

**Comment by SychO9** [comment]
seems to be core check-typings or build-typings or both

**Review by SychO9 — APPROVED** [review]

**Pull request merged**